required/latex-lab/latex-lab-graphic.dtx

```diff
@@ -318,77 +748,307 @@
+% in places where this is wanted. (tagging can not be resumed inside a tagging hook,
+% so has to use some external method).
+%    \begin{macrocode}
+\NewTaggingSocketPlug{graphic/begin}{text}
+  {
+    \tag_mc_end_push:
+    \tag_mc_begin:n{artifact}
+  }
+\NewTaggingSocketPlug{graphic/end}{text}
+  {
+    #2
+    \tag_mc_end:
+    \tag_mc_begin_pop:n{}
+  }
+%    \end{macrocode}
+% \end{plugdecl}
+% 
+% \begin{plugdecl}{off (tagsupport/graphic/begin),off (tagsupport/graphic/end)}
+% This plug can be used for text tagging.
+% It basically does the same as the artifact plug. The main reason that it
+% exist is that it looks more consistend and
+% that we can test for the plug name and so restart tagging
+% in places where this is wanted. (tagging can not be resumed inside a tagging hook,
+% so has to use some external method).
+%    \begin{macrocode}
+\NewTaggingSocketPlug{graphic/begin}{off}{}
+\NewTaggingSocketPlug{graphic/end}{off}{#2}
+%    \end{macrocode}
+% \end{plugdecl}
 
-% \begin{macro}{\Gin@tag@struct@end}
+% 
+%  By default we use the alt plugs
 %    \begin{macrocode}
-\cs_new_protected:Npn\Gin@tag@struct@end
- {
-   \tag_if_active:T
-    {
-     \tag_mc_end:
-     \bool_if:NF\l_@@_graphic_artifact_bool
+\AssignTaggingSocketPlug{graphic/begin}{alt}
+\AssignTaggingSocketPlug{graphic/end}{alt}
+%    \end{macrocode}
+%
+% \begin{plugdecl}{default (tagsupport/graphic/text/begin),
+%  default (tagsupport/picture/text/end)}
+% These sockets are used inside the text
+% plugs and ends the previous mc and restarts it after the text.
+% Not used by \cs{includegraphics}. Unclear if they can be used by
+% the \env{picture} environment.
+%    \begin{macrocode}
+\NewTaggingSocketPlug{graphic/text/begin}{default}
+  {
+   \tag_mc_end:
+   \tag_mc_begin:n{}
+  }
+\NewTaggingSocketPlug{graphic/text/end}{default}
+  {
+   \tag_mc_end:
+   \tag_mc_begin:n{artifact}
+  }
+\AssignTaggingSocketPlug{graphic/text/begin}{default}
+\AssignTaggingSocketPlug{graphic/text/end}{default}
+%    \end{macrocode}
+% \end{plugdecl}
+% 
+% \subsection{Generic keys}
+% The keys are used by picture and \cs{includegraphics} (through the 
+% \texttt{tagging-setup} key
+%    \begin{macrocode}
+\keys_define:nn{tag/graphic}
+   {
+     ,alt .code:n =  
+        {                    
+          \tl_if_empty:nF{#1}
+           { \tl_set:Ne\l_@@_graphic_alt_tl{\text_purify:n{#1}}}
+          \tl_set:Nn\l_tag_graphic_mode_tl{alt}
+        }
+     ,artifact .code:n = 
+        {
+          \tl_set:Nn\l_tag_graphic_mode_tl{artifact}
+        }
+     ,actualtext .code:n = 
+        {
+          \tl_if_empty:nF{#1}
+           { \tl_set:Ne\l_@@_graphic_actual_tl{\text_purify:n{#1}} }          
+          \tl_set:Nn\l_tag_graphic_mode_tl{actualtext}
+        }
+     ,text .code:n =
+        {
+          \tl_set:Nn\l_tag_graphic_mode_tl{text}
+        }   
+     ,off .code:n =
+        {
+          \tl_set:Nn\l_tag_graphic_mode_tl{off}
+        }   
+     ,correct-BBox .code:n = 
```

Comment:
as noted above, I think  `adjust-BBox` or `BBox-correction` would be better names